File: kora-java-crud-submodule/kora-java-crud-submodule-pet-api/src/main/java/ru/tinkoff/kora/example/submodule/pet/PetModule.java
package ru.tinkoff.kora.example.submodule.pet;

import ru.tinkoff.kora.cache.caffeine.CaffeineCacheModule;
import ru.tinkoff.kora.common.KoraSubmodule;
import ru.tinkoff.kora.database.jdbc.JdbcDatabaseModule;
import ru.tinkoff.kora.example.submodule.common.CommonModule;
import ru.tinkoff.kora.resilient.ResilientModule;

@KoraSubmodule
public interface PetModule extends
        CommonModule,
        JdbcDatabaseModule,
        CaffeineCacheModule,
        ResilientModule {

}

File: examples/kotlin/kora-kotlin-crud-submodule/kora-kotlin-crud-submodule-pet-api/src/main/kotlin/ru/tinkoff/kora/kotlin/example/submodule/pet/PetModule.kt
package ru.tinkoff.kora.kotlin.example.submodule.pet

import ru.tinkoff.kora.cache.caffeine.CaffeineCacheModule
import ru.tinkoff.kora.common.KoraSubmodule
import ru.tinkoff.kora.database.jdbc.JdbcDatabaseModule
import ru.tinkoff.kora.kotlin.example.submodule.common.CommonModule
import ru.tinkoff.kora.resilient.ResilientModule

@KoraSubmodule
interface PetModule : CommonModule, JdbcDatabaseModule, CaffeineCacheModule, ResilientModule
